### ISSUE TITLE
Using ptrCast on function pointer fails when assigned to const

### ISSUE THREAD

**Issue by mkeeter** [ziglang/zig#7217]
Here's a minimal example:
```
pub extern fn f() void;

pub fn main() void {
    f(); // This compiles fine, though it will fail at the linker stage

    var g = @ptrCast(fn (c_int) void, f);
    g(1); // This also compiles fine, and will also fail at the linker stage

    const h = @ptrCast(fn (c_int) void, f);
    h(1); // This fails to compile; the only difference is using const rather than var
}
```

The reported error is
```
./test.zig:10:6: error: expected 0 argument(s), found 1
    h(1);
     ^
./test.zig:1:5: note: declared here
pub extern fn f() void;
    ^
```

Zig version `0.7.0+a1fb10b76`

(I know this seems like a wacky thing to do, but it's the [canonical way to call `obj_msgSend`](https://www.mikeash.com/pyblog/objc_msgsends-new-prototype.html) in recent macOS SDKs)

**Comment by daurnimator** [comment]
This looks like a bug to me.

**Comment by mikdusan** [comment]
with latest compiler:
- compile error is gone
- appears this was fixed between 0.9.1 and 0.10.0

<details>
<summary>repro.zig</summary>

```zig
pub extern fn f() void;

pub fn main() void {
    f(); // This compiles fine, though it will fail at the linker stage

    var g = @ptrCast(*const fn (c_int) void, &f);
    g(1); // This also compiles fine, and will also fail at the linker stage

    const h = @ptrCast(*const fn (c_int) void, &f);
    h(1); // This fails to compile; the only difference is using const rather than var
}
```
</details>

```sh
$ uname -a
Linux archlinux 6.1.2-arch1-1 #1 SMP PREEMPT_DYNAMIC Sat, 31 Dec 2022 17:40:35 +0000 x86_64 GNU/Linux

$ zig version
0.11.0-dev.1241+0507ced8c

$ zig build-obj repro.zig
# no stdout/stderr (good)
```
